src/ZodError.ts

```diff
@@ -290,6 +298,11 @@ export const defaultErrorMap = (
     case ZodIssueCode.invalid_union:
       message = `Invalid input`;
       break;
+    case ZodIssueCode.invalid_union_discriminator:
+      message = `Invalid discriminator value. Expected one of: ${issue.expectedOneOf.join(
+        ", "
+      )}. Received ${issue.received}.`;
```

Comment:
As mentioned, I think we want to avoid revealing the input data here. Following the existing example from `invalid_enum_value`:

```suggestion
      message = `Invalid discriminator value. Expected one of: ${issue.expectedOneOf.join(
        ", "
      )}`;
```

Reply:
I removed it from the error message and the issue object. I also updated the tests.

Reply:
And as you mentioned `invalid_enum_value`, I made the `invalid_discriminator_value` issue even more similar to it.